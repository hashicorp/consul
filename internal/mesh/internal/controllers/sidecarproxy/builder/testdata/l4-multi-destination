{
  "proxyState": {
    "clusters": {
      "api-1.default.dc1.internal.foo.consul": {
        "endpointGroup": {
          "dynamic": {
            "config": {
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-1.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api1-identity"
                  ]
                }
              }
            }
          }
        }
      },
      "api-2.default.dc1.internal.foo.consul": {
        "endpointGroup": {
          "dynamic": {
            "config": {
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-2.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api2-identity"
                  ]
                }
              }
            }
          }
        }
      }
    },
    "identity": {
      "name": "test-identity",
      "tenancy": {
        "namespace": "default",
        "partition": "default",
        "peerName": "local"
      }
    },
    "listeners": [
      {
        "direction": "DIRECTION_OUTBOUND",
        "hostPort": {
          "host": "1.1.1.1",
          "port": 1234
        },
        "name": "api-1:tcp:1.1.1.1:1234",
        "routers": [
          {
            "l4": {
              "name": "api-1.default.dc1.internal.foo.consul",
              "statPrefix": "upstream.api-1.default.default.dc1"
            }
          }
        ]
      },
      {
        "direction": "DIRECTION_OUTBOUND",
        "name": "api-2:tcp:/path/to/socket",
        "routers": [
          {
            "l4": {
              "name": "api-2.default.dc1.internal.foo.consul",
              "statPrefix": "upstream.api-2.default.default.dc1"
            }
          }
        ],
        "unixSocket": {
          "mode": "0666",
          "path": "/path/to/socket"
        }
      }
    ]
  },
  "requiredEndpoints": {
    "api-1.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-1",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v1alpha1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    },
    "api-2.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-2",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v1alpha1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    }
  }
}