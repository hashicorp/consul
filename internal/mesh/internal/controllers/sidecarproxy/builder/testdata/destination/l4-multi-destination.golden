{
  "proxyState": {
    "clusters": {
      "null_route_cluster": {
        "endpointGroup": {
          "static": {
            "config": {
              "connectTimeout": "10s"
            }
          }
        },
        "name": "null_route_cluster"
      },
      "tcp.api-1.default.dc1.internal.foo.consul": {
        "altStatName": "tcp.api-1.default.dc1.internal.foo.consul",
        "endpointGroup": {
          "dynamic": {
            "config": {
              "connectTimeout": "5s",
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "alpnProtocols": [
                "consul~tcp"
              ],
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-1.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api1-identity"
                  ],
                  "trustBundlePeerNameKey": "local"
                }
              }
            }
          }
        },
        "name": "tcp.api-1.default.dc1.internal.foo.consul"
      },
      "tcp.api-2.default.dc1.internal.foo.consul": {
        "altStatName": "tcp.api-2.default.dc1.internal.foo.consul",
        "endpointGroup": {
          "dynamic": {
            "config": {
              "connectTimeout": "5s",
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "alpnProtocols": [
                "consul~tcp"
              ],
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-2.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api2-identity"
                  ],
                  "trustBundlePeerNameKey": "local"
                }
              }
            }
          }
        },
        "name": "tcp.api-2.default.dc1.internal.foo.consul"
      },
      "tcp2.api-1.default.dc1.internal.foo.consul": {
        "altStatName": "tcp2.api-1.default.dc1.internal.foo.consul",
        "endpointGroup": {
          "dynamic": {
            "config": {
              "connectTimeout": "5s",
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "alpnProtocols": [
                "consul~tcp2"
              ],
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-1.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api1-identity"
                  ],
                  "trustBundlePeerNameKey": "local"
                }
              }
            }
          }
        },
        "name": "tcp2.api-1.default.dc1.internal.foo.consul"
      },
      "tcp2.api-2.default.dc1.internal.foo.consul": {
        "altStatName": "tcp2.api-2.default.dc1.internal.foo.consul",
        "endpointGroup": {
          "dynamic": {
            "config": {
              "connectTimeout": "5s",
              "disablePanicThreshold": true
            },
            "outboundTls": {
              "alpnProtocols": [
                "consul~tcp2"
              ],
              "outboundMesh": {
                "identityKey": "test-identity",
                "sni": "api-2.default.dc1.internal.foo.consul",
                "validationContext": {
                  "spiffeIds": [
                    "spiffe://foo.consul/ap/default/ns/default/identity/api2-identity"
                  ],
                  "trustBundlePeerNameKey": "local"
                }
              }
            }
          }
        },
        "name": "tcp2.api-2.default.dc1.internal.foo.consul"
      }
    },
    "identity": {
      "name": "test-identity",
      "tenancy": {
        "namespace": "default",
        "partition": "default",
        "peerName": "local"
      },
      "type": {
        "group": "auth",
        "groupVersion": "v2beta1",
        "kind": "WorkloadIdentity"
      }
    },
    "listeners": [
      {
        "direction": "DIRECTION_OUTBOUND",
        "hostPort": {
          "host": "1.1.1.1",
          "port": 1234
        },
        "name": "api-1:tcp:1.1.1.1:1234",
        "routers": [
          {
            "l4": {
              "statPrefix": "upstream.tcp.api-1.default.default.dc1",
              "weightedClusters": {
                "clusters": [
                  {
                    "name": "tcp.api-2.default.dc1.internal.foo.consul",
                    "weight": 60
                  },
                  {
                    "name": "tcp.api-1.default.dc1.internal.foo.consul",
                    "weight": 40
                  },
                  {
                    "name": "null_route_cluster",
                    "weight": 10
                  }
                ]
              }
            }
          }
        ]
      },
      {
        "direction": "DIRECTION_OUTBOUND",
        "name": "api-2:tcp:/path/to/socket",
        "routers": [
          {
            "l4": {
              "cluster": {
                "name": "tcp.api-2.default.dc1.internal.foo.consul"
              },
              "statPrefix": "upstream.tcp.api-2.default.default.dc1"
            }
          }
        ],
        "unixSocket": {
          "mode": "0666",
          "path": "/path/to/socket"
        }
      },
      {
        "direction": "DIRECTION_OUTBOUND",
        "hostPort": {
          "host": "1.1.1.1",
          "port": 2345
        },
        "name": "api-1:tcp2:1.1.1.1:2345",
        "routers": [
          {
            "l4": {
              "statPrefix": "upstream.tcp2.api-1.default.default.dc1",
              "weightedClusters": {
                "clusters": [
                  {
                    "name": "tcp2.api-2.default.dc1.internal.foo.consul",
                    "weight": 60
                  },
                  {
                    "name": "tcp2.api-1.default.dc1.internal.foo.consul",
                    "weight": 40
                  },
                  {
                    "name": "null_route_cluster",
                    "weight": 10
                  }
                ]
              }
            }
          }
        ]
      },
      {
        "direction": "DIRECTION_OUTBOUND",
        "name": "api-2:tcp2:/path/to/socket",
        "routers": [
          {
            "l4": {
              "cluster": {
                "name": "tcp2.api-2.default.dc1.internal.foo.consul"
              },
              "statPrefix": "upstream.tcp2.api-2.default.default.dc1"
            }
          }
        ],
        "unixSocket": {
          "mode": "0666",
          "path": "/path/to/socket"
        }
      }
    ]
  },
  "requiredEndpoints": {
    "tcp.api-1.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-1",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v2beta1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    },
    "tcp.api-2.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-2",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v2beta1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    },
    "tcp2.api-1.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-1",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v2beta1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    },
    "tcp2.api-2.default.dc1.internal.foo.consul": {
      "id": {
        "name": "api-2",
        "tenancy": {
          "namespace": "default",
          "partition": "default",
          "peerName": "local"
        },
        "type": {
          "group": "catalog",
          "groupVersion": "v2beta1",
          "kind": "ServiceEndpoints"
        }
      },
      "port": "mesh"
    }
  },
  "requiredLeafCertificates": {
    "test-identity": {
      "name": "test-identity",
      "namespace": "default",
      "partition": "default"
    }
  },
  "requiredTrustBundles": {
    "local": {
      "peer": "local"
    }
  }
}